src/transformers/quantizers/quantizer_bitnet.py

```diff
@@ -95,6 +101,27 @@ def _process_model_before_weight_loading(
             pre_quantized=self.pre_quantized,
         )
 
+    def adjust_max_memory(
+        self, max_memory: Dict[str, Union[int, str]]
+    ) -> Dict[str, Union[int, str]]:
+        max_memory = {key: val * 0.90 for key, val in max_memory.items()}
+        return max_memory
+
+    def update_torch_dtype(self, torch_dtype: "torch.dtype") -> "torch.dtype":
+        if torch_dtype is None:
+            logger.info(
+                "Overriding torch_dtype=%s with `torch_dtype=torch.bfloat16`"
+                "Pass your own torch_dtype to specify the dtype of the remaining non-linear layers or pass"
+                "torch_dtype=torch.bfloat16 to remove this warning.",
+                torch_dtype,
+            )
+            torch_dtype = torch.bfloat16
+        return torch_dtype
+
+    def adjust_target_dtype(self, target_dtype: "torch.dtype") -> "torch.dtype":
+        target_dtype = torch.int8
```

Comment:
did you check if the modules size were correctly computed if we set this ?

Reply:
Yes seems to be working fine

Reply:
Nice